dataset.py

```diff
@@ -1,77 +1,167 @@
 import os
+import nltk
 import pandas as pd
+import torch
 from torch.utils.data import Dataset, DataLoader
+from torch.nn.utils.rnn import pad_sequence
 from PIL import Image
 from torchvision import transforms
+
+
+class Vocabulary():
```

Comment:
put the class in a different file so is much clear to identify where it's. In general I would suggest a class per file